plugins/catalog-backend/src/ingestion/processors/gitlab/client.ts

```diff
@@ -46,6 +54,70 @@ export class GitLabClient {
     return this.pagedRequest(`${this.config.apiBaseUrl}/projects`, options);
   }
 
+  async getProject(project_id: string | number): Promise<GitLabProject> {
+    if (!this.cache) {
+      return await this._getProject(project_id);
+    }
+    const cacheKey = `gitlabproject--${project_id}`;
+    const res = this.cache.get(cacheKey) as unknown as GitLabProject;
+    if (Object.keys(res).length > 0) {
+      return res;
+    }
+    const output = await this._getProject(project_id);
+    await this.cache.set(cacheKey, output);
+    return output;
+  }
+
+  private async _getProject(
```

Comment:
Just a style nit but we tend to avoid `_` prefixes, could for example call it `fetchProject`